app/scripts/filters/util.js

```diff
@@ -514,4 +514,22 @@ angular.module('openshiftConsole')
     return function(feature) {
       return _.get(Constants, ['ENABLE_TECH_PREVIEW_FEATURE', feature], false);
     };
-  });
+  })
+  .filter('mobileClientIconClass', function() {
+    return function(mobileClientResource) {
+      var clientIcon = _.get(mobileClientResource, 'metadata.labels.icon');
+      var clientType = _.get(mobileClientResource, 'spec.clientType');
+      switch (clientType) {
+        case 'android':
+        case 'iOS':
+          clientIcon = 'fa ' + clientIcon;
```

Comment:
I don't think we should make assumptions about the kind of icon. It would be better if we can have the full icon class like `fa fa-foo` if possible in an annotation.

Reply:
okay I will make the change to the CRD